### ISSUE TITLE
Host API reference docs for cdktf and providers

### ISSUE THREAD

**Issue by jsteinich** [hashicorp/terraform-cdk#748]
> btw, finding the meaning of .fqn function was not straight forward either. Is there api reference guide of the cdktf functions?

Not at this time. The Terraform provider documentation is generally your best place to look; however, I realize that doesn't cover the few built in pieces like `fqn`. It's been brought up a couple times now, so I'll make a separate issue to track.

_Originally posted by @jsteinich in https://github.com/hashicorp/terraform-cdk/issues/374#issuecomment-851032632_

**Comment by jsteinich** [comment]
There was also a slack thread which mentioned having API documentation for the pre-built providers. Apparently there is a markdown file, but it is too large for GitHub to display.

**Comment by skorfmann** [comment]
> but it is too large for GitHub to display

Yes, for large providers like AWS it's not really usable (11 MB file). However,  I think we'll start with docs for cdktf itself and then likely expand to the pre-built providers. Making the tooling more accessible via docs is certainly very important.

**Comment by xiehan** [comment]
I'm going to close this ticket because:
- We do have an [API Reference](https://developer.hashicorp.com/terraform/cdktf/api-reference/overview) now for CDKTF itself
- We also have API reference docs for all prebuilt providers via Construct Hub (even if the quality is not the greatest)
- We are slowly but surely working on #1298 which will eventually capture all other providers

If there are more specific asks or improvements, please open new tickets to capture each of those ideas; smaller and more specific tickets are easier for us to prioritize.
